sdk/compatability/src/lib.rs

```diff
@@ -0,0 +1,694 @@
+//! A suite of client/sdk compatability tests
+
+#![cfg(test)]
+
+use anyhow::Result;
+use diem_sdk::{
+    crypto::HashValue,
+    transaction_builder::{self, ScriptCall},
+    types::{account_address::AccountAddress, transaction::Script},
+};
+
+mod env;
+use env::DualAttestationMessage;
+pub use env::{Coffer, Currency, Environment};
+
+mod account;
+pub use account::LocalAccount;
+
+#[test]
+#[ignore]
+fn metadata() -> Result<()> {
+    let env = Environment::from_env();
+    let client = env.client();
+
+    let metadata = client.get_metadata()?.into_inner();
+
+    // get_metadata documentation states that the following fields will be present when no version
+    // argument is provided
+    metadata.script_hash_allow_list.unwrap();
+    metadata.diem_version.unwrap();
+    metadata.module_publishing_allowed.unwrap();
+    metadata.dual_attestation_limit.unwrap();
+
+    Ok(())
+}
```

Comment:
Nope there should be zero issues since these are all "Public Usage" tests in that they are from the perspective of a user of the blockchain. There's a lot more that we need to do on the testing front (especially compatibility) and we should probably look into creating a set of "admin usage" tests that are executed from the perspective of a system like AOS. Those types of tests would need to executed serially or with their own networks entirely.